lifecycle.Rmd

```diff
@@ -65,78 +264,193 @@ usethis::use_version()
 #> Selection: 
 ```
 
-## Version number {#sec-release-version}
+## Backward compatibility {#sec-lifecycle-backward-compatibility}
+
+The big difference between major and minor releases is whether or not the code is backward compatible.
+This difference is a bit academic in the R community, because of the way most users manage package installation.
+If we're being honest, most R users don't manage package versions in a very intentional way.
+Given the way `update.packages()` and `install.packages()` work, it's quite easy to upgrade a package to a new major version without really meaning to, especially for dependencies of the target package.
+This, in turn, can lead to unexpected exposure to breaking changes in code that previously worked.
+This unpleasantness has implications both for users and for maintainers.
+
+If it's important to protect a data product against change in its R package dependencies, we recommend the use of a project-specific package library.
+In particular, we like to implement this approach using the [renv package](https://rstudio.github.io/renv/).
+This supports a lifestyle where a user's default package library is managed in the usual, somewhat haphazard way.
+But any project that has a specific, higher requirement for reproducibility is managed with renv.
+This keeps package updates triggered by work in project A from breaking the code in project B and also helps with collaboration and deployment.
+
+We suspect that package-specific libraries and tools like renv are currently under-utilized in the R world.
+That is, lots of R users still use just one package library.
+Therefore, package maintainers still need to exercise considerable caution and care when they introduce breaking changes, regardless of what's happening with the version number.
+In section ??,
+we describe how tidyverse packages approach this, supported by tools in the lifecycle package.
+
+As with dependencies (@sec-dependencies-pros-cons), we find that extremism isn't a very productive stance.
+Extreme resistance to breaking change puts a significant drag on ongoing development and maintenance.
+Backward compatible code tends to be harder to work with because of the need to maintain multiple paths to support functionality from previous versions.
+The harder you strive to maintain backward compatibility, the harder it is to develop new features or fix old mistakes.
+This, in turn, can discourage adoption by new users and can make it harder to recruit new contributors.
+Be concerned about backward compatibility, but don't let it paralyze you.
```

Comment:
OTOH if you constantly make breaking changes, people will find it very frustrating to use your package and possibly stop. Find a happy medium.

Reply:
Added